src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostOptions.cs

```diff
@@ -29,5 +29,22 @@ internal static class RemoteHostOptions
 
         [ExportOption]
         public static readonly Option<bool> RemoteHostTest = new Option<bool>(nameof(InternalFeatureOnOffOptions), nameof(RemoteHostTest), defaultValue: false);
+
+        public const int CrashVS = 0;
+        public const int ShowInfoBarAndRunFeaturesInProcIfPossible = 1;
+        public const int ShowInfoBarAndDisableAllFeatures = 2;
+
+        // Define VS behavior on unintentional OOP shutdown such as user killing OOP process
+        //
+        // CrashVS: 
+        //      crash VS
+        // ShowInfoBarAndRunFeaturesInProcIfPossible: [default]
+        //      show info bar and let VS keep running. some features might keep running inproc based on how they are implemented.
```

Comment:
>all features who want to run in OOP need to write their features in a way that knows how to handle OOP killed case.

How would a feature do this?  How does a feature know if OOP has been killed?

> I made these options since Matt asked me until shiproom choose one of those.

That's fine.  I'm talking about what i think an appropriate default will be.

The problem with running in proc is that the experience might get really poor.  Users may report this, and we'll not know how to repro getting into the state that they're in.

It also means we have to do much more perf testing to validate the experience both for in-proc and out-of-proc.  Is this going to be done?